domain/storageprovisioning/state/base_test.go

```diff
@@ -254,6 +256,38 @@ VALUES (?, ?, ?, ?, 0, 100, 'rootfs')
 	return storageInstanceUUID
 }
 
+func (s *baseSuite) getStorageID(
+	c *tc.C, storageInstanceUUID domainstorage.StorageInstanceUUID,
+) string {
+	var storageID string
+	err := s.DB().QueryRowContext(
+		c.Context(), `
+SELECT storage_id FROM storage_instance WHERE uuid = ?`,
+		storageInstanceUUID.String(),
+	).Scan(&storageID)
+	c.Assert(err, tc.ErrorIsNil)
+	return storageID
+}
+
+func (s *baseSuite) newStorageAttachment(
+	c *tc.C,
+	storageInstanceUUID domainstorage.StorageInstanceUUID,
+	unitUUID coreunit.UUID,
+	life domainlife.Life,
+) {
+	err := s.TxnRunner().StdTxn(
+		c.Context(),
+		func(ctx context.Context, tx *sql.Tx) error {
+			_, err := tx.ExecContext(ctx, `
+INSERT INTO storage_attachment (storage_instance_uuid, unit_uuid, life_id)
+VALUES (?, ?, ?)
+`, storageInstanceUUID.String(), unitUUID.String(), life)
```

Comment:
Also for test helpers, just run everything outside a transaction unless you actually need it.